### PR TITLE
Book task 10 category

### PR DESCRIPTION
**Pull request by MolchanovAlexander**

### CODE REVIEW

**Review by ValentynSharshon — APPROVED** [review]
LGFM, good job!

File: src/main/java/com/example/bookstorewebapp/model/Book.java [hunk 1]
```diff
@@ -32,6 +38,13 @@ public class Book {
     private String coverImage;
     @Column(nullable = false)
     private boolean isDeleted = false;
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstorewebapp/model/Book.java]:
```suggestion
    @ManyToMany(fetch = FetchType.LAZY)
```

File: src/main/java/com/example/bookstorewebapp/service/book/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -19,11 +23,28 @@
...
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final CategoryRepository categoryRepository;
     private final BookMapper bookMapper;
     private final BookSpecificationBuilder specificationBuilder;
 
+    @Override
+    public List<BookDtoWithoutCategories> findAllByCategoryId(Long id, Pageable pageable) {
+        if (!categoryRepository.existsById(id)) {
+            throw new EntityNotFoundException("There is no book to update with id: " + id);
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstorewebapp/service/book/impl/BookServiceImpl.java]: I believe error message should be different

File: src/main/java/com/example/bookstorewebapp/service/category/impl/CategoryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                () -> new EntityNotFoundException("Can't find category by id: " + id)
+        );
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        if (!categoryRepository.existsById(id)) {
+            throw new EntityNotFoundException("There is no category to delete with id: " + id);
+        }
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstorewebapp/service/category/impl/CategoryServiceImpl.java]: I suggest to move this check to separate method and reuse in delete and update. You should also remake error message like "There is no category with id: "

File: src/main/resources/db/changelog/changes/05-create-users-roles-table.yaml [hunk 4]
```diff
@@ -11,43 +11,53 @@ databaseChangeLog:
...
             columns:
               - column:
                   name: user_id
-                  value: 2
+                  valueComputed: (SELECT id FROM users WHERE email = 'alice@alice.com')
               - column:
                   name: role_id
-                  value: 1
+                  valueComputed: (SELECT id FROM roles WHERE role = 'USER')
+
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/05-create-users-roles-table.yaml]:
```suggestion
```

File: src/main/resources/db/changelog/changes/06-create-categories-table.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+                      unique: true
+                - column:
+                    name: description
+                    type: varchar(255)
+                - column:
+                    name: is_deleted
+                    type: boolean
+                    defaultValueBoolean: false
+                    constraints:
+                      nullable: false
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/06-create-categories-table.yaml]:
```suggestion
                      nullable: false
                      
```

File: src/main/resources/db/changelog/changes/07-create-books-categories-table.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            referencedColumnNames: id
+        - addForeignKeyConstraint:
+            baseTableName: books_categories
+            baseColumnNames: category_id
+            constraintName: fk_category_id
+            referencedTableName: categories
+            referencedColumnNames: id
+        - addPrimaryKey:
+            tableName: books_categories
+            columnNames: book_id, category_id
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/07-create-books-categories-table.yaml]:
```suggestion
            columnNames: book_id, category_id
            
```

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good job! Left some minor comments

**Pull request merged**